pyrovision/datasets/utils.py

```diff
@@ -3,153 +3,34 @@
-        url, filename = url
-
-    if not isinstance(url, str):
-        raise TypeError('expected argument url to be of type <str>')
-
-    # Root folder
-    root = Path(root).expanduser()
-    root.mkdir(parents=True, exist_ok=True)
-    if not filename:
-        filename = get_fname(url)
-
-    fpath = root.joinpath(filename)
-
-    # Download file
-    if check_integrity(fpath, md5):
-        if verbose:
-            print(f'Using downloaded and verified file: {fpath}')
-    else:
-        success = False
-        # Allow multiple retries
-        for idx in range(retries + 1):
-            try:
-                url_retrieve(url, fpath, timeout)
-                success = True
-            except Exception as e:
-                # Try switching to http
-                if url.startswith('https'):
-                    try:
-                        url_retrieve(url.replace('https:', 'http:'), fpath, timeout)
-                        success = True
-                    except Exception:
-                        success = False
-                # Handle exception
-                if not success and (idx == retries):
-                    if not silent:
-                        raise e
-                    elif verbose:
-                        print(e)
-            if success:
-                break
-
-
-def parallel(func, arr, threads=None, leave=False):
-    """Download a file accessible via URL with mutiple retries
-
-    Args:
-        func (callable): function to be executed on multiple workers
-        arr (iterable): function argument's values
-        threads (int, optional): number of workers to be used for multiprocessing
-        leave (bool, optional): whether traces of progressbar should be kept upon termination
-
-    Returns:
-        list: list of function's results
-    """
-
-    if threads is None:
-        threads = min(16, mp.cpu_count())
-    if threads < 2:
-        results = [func(arg) for arg in tqdm(arr, total=len(arr), leave=leave)]
-    else:
-        with ThreadPool(threads) as tp:
-            results = list(tqdm(tp.imap_unordered(func, arr), total=len(arr)))
-    if any([o is not None for o in results]):
-        return results
-
-
-def download_urls(entries, root, timeout=4, retries=4, threads=None, silent=True):
-    """Download multiple URLs a file accessible via URL with mutiple retries
-
-    Args:
-        entries (list<str, str>): URL and destination filen
-        root (pathlib.Path): folder where the files will be saved in
-        timeout (float, optional): number of seconds before the request times out
-        retries (int, optional): number of additional allowed download attempts
-        threads (int, optional): number of threads to be used for multiprocessing
-        silent (bool, optional): whether Exception should be raised upon download failure
-    """
-
-    parallel(partial(download_url, root=root, timeout=timeout, retries=retries, silent=silent),
-             entries, threads=threads)
```

Comment:
Even if OpenFire doesn't use this anymore, this could still be quite useful in the future (downloading list of files in parallel is much faster than doing this sequentially), and I don't see many reasons to remove this :sweat_smile:

Reply:
It was not used anymore, but I can leave it if you want :)  Just I need to setup a "fake test" for it since it's not tested in OpenFire anymore